source/Ship.cpp

```diff
@@ -464,6 +463,14 @@ void Ship::Load(const DataNode &node)
 			description += child.Token(1);
 			description += '\n';
 		}
+		else if(!variantName.empty()) {
+			if ((key == "remove") && (child.Size() >= 2)) {
+				if (child.Token(1) == "bays")
+					removeBaseFeature |= (1 << REMOVE_SET_BAYS);
+			}
+			else
+				child.PrintTrace("Skipping unrecognized attribute:");
+		}
```

Comment:
```suggestion
		else if( key == "remove" && child.Size() >= 2 && child.Token(1) == "bays") {
			if(!variantName.empty())
				removeBaseFeature |= (1 << REMOVE_SET_BAYS);
			else
				child.PrintTrace("Skipping invalid 'remove bays' on a non variant:");
		}
```
If we're going to have a separate error message from the rest, may as well make it specific!
Other than that I just like it better in one line.

Reply:
fixed in latest commit